### ISSUE TITLE
curl_easy_reset does not properly reset handle when switching from AUTH_DIGEST to AUTH_BEARER

### ISSUE THREAD

**Issue by delperugia** [curl/curl#5901]
### I did this
I make two requests re-using the same easy handle, with a reset of the handle in between. First request uses auth Digest authorization, the second Bearer. Second request gets a 401, while if I instead makes a new handle for the second request it successfully gets a 200.

It looks similar to #2255 but I use version 7.68.0 where it is fixed. Attached a test program reproducing this problem.
[curl_reset.zip](https://github.com/curl/curl/files/5156259/curl_reset.zip)

When executed without a argument, it creates a new handle for the 2nd request and works:
```
~$ ./a.out
libcurl/7.68.0 OpenSSL/1.1.1f zlib/1.2.11 brotli/1.0.7 libidn2/2.2.0 libpsl/0.21.0 (+libidn2/2.2.0) libssh/0.9.3/openssl/zlib nghttp2/1.40.0 librtmp/2.3
Digest:
{
  "authenticated": true,
  "user": "jane"
}
Bearer:
{
  "authenticated": true,
  "token": "secret"
}
```

When executed with an argument, calling **curl_easy_reset**, 2nd request fails:
```
~$ ./a.out 1
libcurl/7.68.0 OpenSSL/1.1.1f zlib/1.2.11 brotli/1.0.7 libidn2/2.2.0 libpsl/0.21.0 (+libidn2/2.2.0) libssh/0.9.3/openssl/zlib nghttp2/1.40.0 librtmp/2.3
Digest:
{
  "authenticated": true,
  "user": "jane"
}
Bearer:
```

Wireshark capture shows that the **Authorization** header is not set in that case:
```
GET /bearer HTTP/1.1
Host: httpbin.org
Accept: */*

HTTP/1.1 401 UNAUTHORIZED
....
```

### I expected the following
Second request to succeed.

### curl/libcurl version
curl 7.68.0 (x86_64-pc-linux-gnu) libcurl/7.68.0 OpenSSL/1.1.1f zlib/1.2.11 brotli/1.0.7 libidn2/2.2.0 libpsl/0.21.0 (+libidn2/2.2.0) libssh/0.9.3/openssl/zlib nghttp2/1.40.0 librtmp/2.3
Release-Date: 2020-01-08
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtmp rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: AsynchDNS brotli GSS-API HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile libz NTLM NTLM_WB PSL SPNEGO SSL TLS-SRP UnixSockets

### operating system
Ubuntu 20.04.1 LTS on WSL2

Linux DESKTOP-2QDKMHN 4.19.104-microsoft-standard #1 SMP Wed Feb 19 06:37:35 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

**Comment by bagder** [comment]
Thanks. When I run your test program with curl 7.72.0 from debian and from current git, I can't reproduce. I get this:

## Debian
~~~
$ ./tryit 1
libcurl/7.72.0 OpenSSL/1.1.1g zlib/1.2.11 brotli/1.0.7 libidn2/2.3.0 libpsl/0.21.0 (+libidn2/2.3.0) libssh2/1.8.0 nghttp2/1.41.0 librtmp/2.3
Digest:
{
  "authenticated": true, 
  "user": "jane"
}
Bearer:
{
  "authenticated": true, 
  "token": "secret"
}
~~~
## git version
~~~
$ ./debugit 1
libcurl/7.73.0-DEV OpenSSL/1.1.1g zlib/1.2.11 brotli/1.0.7 zstd/1.4.5 c-ares/1.16.1 libidn2/2.3.0 libpsl/0.21.0 (+libidn2/2.3.0) libssh2/1.9.0_DEV nghttp2/1.41.0-DEV librtmp/2.3
Digest:
{
  "authenticated": true, 
  "user": "jane"
}
Bearer:
{
  "authenticated": true, 
  "token": "secret"
}

**Comment by delperugia** [comment]
I download, compile and tested versions 7.73.0-DEV, 7.69.0-DEV and 7.68.0-DEV. 7.69 and 7.73 work fine. I can reproduce the problem with 7.68. So it appears it was fixed in 7.69. I guess it is linked to #4824 and can be closed?

**Comment by bagder** [comment]
Does indeed. Thanks, case closed.
